dsgrid/config/project_config.py

```diff
@@ -436,6 +447,30 @@ def get_base_to_supplemental_dimension_mappings_by_types(self, dimension_type: D
             if x.model.from_dimension.dimension_type == dimension_type
         ]
 
+    def get_base_to_supplemental_mapping_records(self, query_name: str):
+        """Return the project's base-to-supplemental dimension mapping records.
+
+        Parameters
+        ----------
+        query_name : str
+
+        Returns
+        -------
+        pyspark.sql.DataFrame
+
+        """
+        dim = self.get_dimension(query_name)
+        dimension_type = dim.model.dimension_type
+        base_dim = self.get_base_dimension(dimension_type)
```

Comment:
This is the same as the following right?
```
if query_name in self.get_base_dimension_query_names():
    raise DSGInvalidParameter(f"Cannot pass base dimension: {dimension_type}/{query_name}")
```

Reply:
That produces the same result, but what I have is more efficient code-wise. You example has to iterate across each dimension.